migration.md

```diff
@@ -1,5 +1,57 @@
 # Migration
 
+## 0.5.0
+
+### Package exports
+
+[Information about Package entry points](https://nodejs.org/api/packages.html#package-entry-points)
+
+We now have a package exports entry in our `package.json`.
+This is to add better support for SSR which uses Node.js module resolution rather than the browser's algorithm.
+One example is that in Node.js a path like `foo/bar` doesn't automatically resolve to `foo/bar/index.js`.
+
+Another big advantage of package exports is pinning down explicitly where consumers can import from, increasing reliability and making explicit our public and private APIs. We can now change our file locations or edit filenames without being worried that some consumer was relying on it through an import we did not intend.
+
+For named imports, use:
+
+```js
+import { Foo } from '@divriots/starter-simba';
+```
+
+For named imports from specific simba packages, use:
+
+```js
+import { Bar } from '@divriots/starter-simba/input';
+```
+
+For custom elements definitions, use:
+
+```js
+import '@divriots/starter-simba/input/define';
+```
+
+When a package exports multiple custom elements, the `define` entrypoint will define all of them. 
+In case you only need a specific one, for example a `checkbox` but not the `checkbox-group` or `checkbox-indeterminate`, use:
+
+```js
+import '@divriots/starter-simba/checkbox/define-checkbox';
```

Comment:
`/define-checkbox';` ?

Reply:
Yep this is right.

```
|- checkbox
|   |- index.js
|   |- define.js
|   |- simba-checkbox.js
|   |- simba-checkbox-group.js
|   |- simba-checkbox-indeterminate.js
```
define.js imports the 3 files below it to define all of the 3 CEs at the same time, but sometimes a user only wants a checkbox and doesn't need the group or indeterminate, in that case you wanna save bytes.

```js
import '@divriots/starter-simba/checkbox/define'; // imports and defines all 3 CEs
import '@divriots/starter-simba/checkbox/define-checkbox'; // only imports and defines the simba-checkbox element
import { SimbaCheckbox } from '@divriots/starter-simba/checkbox'; // gives you the class without polluting the CE registry, e.g. for extending purposes.
```

Reply:
can we make a convention for this?
I'm thinkin of solving 2 problems:
1. define all when all are needed for simplicity
2. allow defining individual ones if not all are needed in rare cases

so for packages with 1 or more component it will look like that

- package-name/define.js (for simplicity I don't want to call it `define-all`, so that in packages with 1 component the name makes sense and you don't need to bother or refactor if more component are added in the future)
- package-name/define-my-component1.js
- package-name/define-my-component2.js

another variation:
- package-name/define.js
- package-name/my-component1.define.js
- package-name/my-component2.define.js

not sure which version is better, but I saw the latter more often

Reply:
I thought if you just released a breaking change, it's better to make another one sooner than later, so people can just skip the current one :D
but in any case, people don't depend on `starter-simba` in their projects, so I gues it's not an issue to make as many breaking changes as needed, because it's a living being that does't need to be as stable as a final DS + we need to innovate more in this field, so why limit ourselves?

about "define" vs "register": I'm totally for "register", I have no idea why "define" appeared in Lion and other libraries, happened without me, so I was just basing my suggestion on it, but if I was designing from scratch, I'd go for "register" and I prefer actually the 2nd variant making it

- `import '@divriots/starter-name/package-name/register.js';`
- `import '@divriots/starter-name/package-name/my-component1.register.js';`
- `import '@divriots/starter-name/package-name/my-component2.register.js';`

why prefer 2 vs 1? 2 reasons:

1. autocomplete, because you can start typing by a component name, it's more straighforward
2. it's a common situation when you have multiple files like

    - `package-name/my-component1.css` or `package-name/my-component1.styles.js`
    - `package-name/my-component1.html` or `package-name/my-component1.template.js`
    - `package-name/my-component1.register.js`
    
    So it makes sense to have all these files named in the same manner. But I see that from the public API it's not that important, however reasons 1 is still a thing there.

Reply:
ha, actually I was wrong about "register", ignore this idea in the previous message, I'll clean this up
"define" is an actual name from the `customElements.define`, so it makes sense